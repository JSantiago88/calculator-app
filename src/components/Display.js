import React from 'react';
import PropTypes from 'prop-types';

const Display = ({ initialValue = [0] }) => {

    return (
        <div className='bg-display'>
            <h1> {initialValue}</h1>
        </div>
    );
};


Display.propTypes = {
    initialValue: PropTypes.array.isRequired
};

export default Display;