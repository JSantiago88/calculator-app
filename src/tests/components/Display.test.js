import React from 'react';
import { shallow } from 'enzyme';

import Display from '../../components/Display';

describe('Test in <Display />', () => {

    const initialValue = [5];
    const wrapper = shallow(<Display initialValue={initialValue} />);

    test('should Display the component correctly', () => {

        expect(wrapper).toMatchSnapshot();
    });

    test('should have a text in the h1', () => {
        const h1 = wrapper.find('h1');

        expect(h1.text().trim()).toBe(initialValue[0].toString());
    });

    test('should have a css class in the div', () => {
        const div = wrapper.find('div');
        const className = div.prop('className');

        expect(className.includes('bg-display')).toBe(true);
    });
});
