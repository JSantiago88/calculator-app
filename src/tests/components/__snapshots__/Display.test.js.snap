// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <Display /> should Display the component correctly 1`] = `
<div
  className="bg-display"
>
  <h1>
     
    5
  </h1>
</div>
`;
